apps/design/backend/src/app.ts

```diff
@@ -335,6 +339,55 @@ function buildApi({ auth, workspace, translator }: AppContext) {
       await store.updateBallotLanguageCodes(electionId, languageCodes);
     },
 
+    async listDistricts(input: {
+      electionId: ElectionId;
+    }): Promise<readonly District[]> {
+      const { election } = await store.getElection(input.electionId);
+      return election.districts;
+    },
+
+    async createDistrict(input: {
+      electionId: ElectionId;
+      newDistrict: District;
+    }): Promise<void> {
+      const district = unsafeParse(DistrictSchema, input.newDistrict);
+      const { election } = await store.getElection(input.electionId);
+      await store.updateElection(input.electionId, {
+        ...election,
+        districts: [...election.districts, district],
+      });
+    },
+
+    async updateDistrict(input: {
+      electionId: ElectionId;
+      districtId: DistrictId;
+      updatedDistrict: Omit<District, 'id'>;
+    }): Promise<void> {
+      const district = unsafeParse(DistrictSchema, {
+        ...input.updatedDistrict,
+        id: input.districtId,
+      });
+      const { election } = await store.getElection(input.electionId);
+      assert(
+        election.districts.some((d) => d.id === input.districtId),
+        'District not found'
+      );
+      const updatedDistricts = election.districts.map((d) =>
+        d.id === input.districtId ? district : d
+      );
+      await store.updateElection(input.electionId, {
```

Comment:
Should we move this operation into a `Store::updateDistrict` method? It's a little odd for this RPC method to use `updateElection` and the `deleteDistrict` one to use `Store::deleteDistrict`. Moving toward having an abstraction around that will enable us to move toward having dedicated tables for districts/precincts/etc as you suggested earlier.

Reply:
I agree that's the eventual direction we should go. I don't see a strong reason to do it now, but also don't have a strong reason not to do it now haha. So I'm down to do it now.

To be clear, under the hood `Store::updateDistrict` would be exactly the same code as is currently in `App::updateDistrict`. Is that what you're imagining too?

Reply:
174e6f3af